### PR TITLE
multi-region: conditional bootstrap

### PR DESCRIPTION
**Pull request by bernadinm**
This adds control to enable a bootstrap node on the remote region

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 output "public_ip" {
   description = "List of public ip addresses created by this module"
-  value       = "${module.dcos-bootstrap-instance.public_ips[0]}"
+  value       = "${element(concat(module.dcos-bootstrap-instance.public_ips, list("")), 0)}"
 }
 
 output "private_ip" {
   description = "List of private ip addresses created by this module"
-  value       = "${module.dcos-bootstrap-instance.private_ips[0]}"
+  value       = "${element(concat(module.dcos-bootstrap-instance.private_ips, list("")), 0)}"
```
Comment (fatz) [inline outputs.tf:13]: oh I see the concat is meant to be used while destroying... do I get this right?

**Review by fatz — COMMENTED** [review]
🤔 why is this needed for molti-region. To me this change will introduce multi AZ but not multi region

File: outputs.tf (hunk 1 above)
Comment (bernadinm) [reply outputs.tf:13]: when the bootstrap is conditionally created in the case of the multi-region environment where it pull the bootstrap from a remote region, having an empty object in an array throws an error. More info on this here: https://github.com/hashicorp/hil/issues/50

The workaround in terraform is to always add a list("") and have it return that instead in the event an empty list exist.

Comment (bernadinm) [reply outputs.tf:13]: And this specific comment as well https://github.com/hashicorp/terraform/issues/16580#issuecomment-342573652

**Review by fatz — APPROVED** [review]
lgtm

**Pull request merged**